ledger/participant-integration-api/src/main/scala/platform/store/backend/DBDTOV1.scala

```diff
@@ -75,67 +80,63 @@ object DBDTOV1 {
       event_sequential_id: Long,
   ) extends DBDTOV1
 
-  // TODO append-only: wartremover complained about having Array-s in case classes. I would prefer case classes. can we work that somehow around? Similarly in other DTO cases...
-  // TODO append-only: there are some options:
-  //   - mixing in SomeArrayEquals if we need array equality for some reason: would be proper if we move SomeArrayEquals out from speedy codebase to scalalib first.
-  //   - spawning somewhere something like trait NeverEqualsOverride { override equals(o: Object): Boolean = false }, and mixing in these classes
-  class ConfigurationEntry(
-      val ledger_offset: String,
-      val recorded_at: Instant,
-      val submission_id: String,
-      val typ: String,
-      val configuration: Array[Byte],
-      val rejection_reason: Option[String],
+  final case class ConfigurationEntry(
+      ledger_offset: String,
+      recorded_at: Instant,
+      submission_id: String,
+      typ: String,
+      configuration: Array[Byte],
+      rejection_reason: Option[String],
   ) extends DBDTOV1
 
-  class PackageEntry(
-      val ledger_offset: String,
-      val recorded_at: Instant,
-      val submission_id: Option[String],
-      val typ: String,
-      val rejection_reason: Option[String],
+  final case class PackageEntry(
+      ledger_offset: String,
+      recorded_at: Instant,
+      submission_id: Option[String],
+      typ: String,
+      rejection_reason: Option[String],
   ) extends DBDTOV1
 
-  class Package(
-      val package_id: String,
-      val upload_id: String,
-      val source_description: Option[String],
-      val size: Long,
-      val known_since: Instant,
-      val ledger_offset: String,
-      val _package: Array[Byte],
+  final case class Package(
+      package_id: String,
+      upload_id: String,
+      source_description: Option[String],
+      size: Long,
+      known_since: Instant,
+      ledger_offset: String,
+      _package: Array[Byte],
```

Comment:
why with `_`?

Reply:
try it without :trollface: :)
I think because it looks better then ``package`` :thinking: but that is also debatable :)
FYI: not part of this review

Reply:
Ah reserved word. But still, first thing that comes to mind is an internal `var` that should not be exposed. Looks confusing. Maybe `packagePayload` would help